scripts/two-node-local-net.sh

```diff
@@ -0,0 +1,105 @@
+#!/usr/bin/env bash
+
+# run a two node local net
+
+set -e
+
+PROJECT_ROOT=$(git rev-parse --show-toplevel)
+# shellcheck disable=SC1090
+source "$(dirname "$0")"/common.sh
+
+cd "$PROJECT_ROOT"
+
+last_modified_rust_file=$(
+  find . -path ./target -prune -o -type f -name '*.rs' -printf '%T@ %p\n' |
+  sort -nr |
+  head -1 |
+  cut -d' ' -f2-
+)
+
+polkadot="target/release/polkadot"
+
+# ensure the polkadot binary exists and is up to date
+if [ ! -x "$polkadot" ] || [ "$polkadot" -ot "$last_modified_rust_file" ]; then
+  cargo build --release
+fi
+
+# setup variables
+node_offset=0
+declare -a node_pids
+declare -a node_pipes
+
+# create a sed expression which injects the node name and stream type into each line
+function make_sed_expr() {
+  name="$1"
+  type="$2"
+
+  printf "s/^/%8s %s: /" "$name" "$type"
+}
+
+# start a node and name its output
+function run_node() {
+  name="$1"
+  # create a named pipe so we can get the node's PID while also sedding its output
+  local stdout
+  local stderr
+  stdout=$(mktemp --dry-run --tmpdir)
+  stderr=$(mktemp --dry-run --tmpdir)
+  mkfifo "$stdout"
+  mkfifo "$stderr"
+  node_pipes+=("$stdout")
+  node_pipes+=("$stderr")
+
+  # compute offsets
+  local port=$((30333+node_offset))
+  local rpc_port=$((9933+node_offset))
+  local ws_port=$((9944+node_offset))
+  node_offset++
+
+  # start the node
+  # TODO: do we need a custom chainspec? Probably.
+  "$polkadot" \
+    --chain polkadot \
```

Comment:
The intent is to run a polkadot net, identical to the polkadot software, but not actually connect to the real Polkadot network. I suspect that we need a custom chainspec here, but I don't yet know what will be precisely required.